From: Steve Langasek <steve.langasek@canonical.com>
Date: Sat, 16 Jul 2011 13:12:48 -0700
Subject: gio: add fallback directory for pre-multiarch compatibility

Origin: vendor, Ubuntu
Bug-Debian: https://bugs.debian.org/634099
Forwarded: no, specific to Debian and its derivatives
---
 gio/giomodule.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gio/giomodule.c b/gio/giomodule.c
index 36c0cef..9f5f6ef 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1149,6 +1149,9 @@ _g_io_modules_ensure_loaded (void)
 
       g_io_module_scope_free (scope);
 
+      /* Hard-coded fallback directory for pre-multiarch compatibility */
+      g_io_modules_scan_all_in_directory ("/usr/lib/gio/modules");
+
       /* Initialize types from built-in "modules" */
       g_type_ensure (g_null_settings_backend_get_type ());
       g_type_ensure (g_memory_settings_backend_get_type ());
