diff -ruN glib-2.6.5.old/tests/Makefile.am glib-2.6.5/tests/Makefile.am
--- glib-2.6.5.old/tests/Makefile.am	2005-02-08 15:15:16.000000000 +0000
+++ glib-2.6.5/tests/Makefile.am	2005-06-10 18:48:34.000000000 +0000
@@ -9,10 +9,10 @@
 libgmodule = $(top_builddir)/gmodule/libgmodule-2.0.la
 libgobject = $(top_builddir)/gobject/libgobject-2.0.la
 
-if PLATFORM_WIN32
 libadd_libgmodule = $(libgmodule)
 libadd_libglib = $(libglib)
 no_undefined = -no-undefined
+if PLATFORM_WIN32
 
 module_test_exp = module-test.exp
 
diff -ruN glib-2.6.5.old/tests/Makefile.in glib-2.6.5/tests/Makefile.in
--- glib-2.6.5.old/tests/Makefile.in	2005-06-10 16:18:46.000000000 +0000
+++ glib-2.6.5/tests/Makefile.in	2005-06-10 18:48:59.000000000 +0000
@@ -249,9 +249,9 @@
 libgmodule = $(top_builddir)/gmodule/libgmodule-2.0.la
 libgobject = $(top_builddir)/gobject/libgobject-2.0.la
 
-@PLATFORM_WIN32_TRUE@libadd_libgmodule = $(libgmodule)
-@PLATFORM_WIN32_TRUE@libadd_libglib = $(libglib)
-@PLATFORM_WIN32_TRUE@no_undefined = -no-undefined
+libadd_libgmodule = $(libgmodule)
+libadd_libglib = $(libglib)
+no_undefined = -no-undefined
 
 @PLATFORM_WIN32_TRUE@module_test_exp = module-test.exp
 
