diff -ruN glib-2.4.7.old/tests/Makefile.am glib-2.4.7/tests/Makefile.am
--- glib-2.4.7.old/tests/Makefile.am	2004-03-06 08:37:46.000000000 +0100
+++ glib-2.4.7/tests/Makefile.am	2004-10-08 22:25:10.000000000 +0200
@@ -9,9 +9,9 @@
 libgmodule = $(top_builddir)/gmodule/libgmodule-2.0.la
 libgobject = $(top_builddir)/gobject/libgobject-2.0.la
 
-if PLATFORM_WIN32
 libadd_libgmodule = $(libgmodule)
 libadd_libglib = $(libglib)
+if PLATFORM_WIN32
 no_undefined = -no-undefined
 
 module_test_exp = module-test.exp
diff -ruN glib-2.4.7.old/tests/Makefile.in glib-2.4.7/tests/Makefile.in
--- glib-2.4.7.old/tests/Makefile.in	2004-10-08 19:59:04.000000000 +0200
+++ glib-2.4.7/tests/Makefile.in	2004-10-08 22:25:25.000000000 +0200
@@ -234,8 +234,8 @@
 libgmodule = $(top_builddir)/gmodule/libgmodule-2.0.la
 libgobject = $(top_builddir)/gobject/libgobject-2.0.la
 
-@PLATFORM_WIN32_TRUE@libadd_libgmodule = $(libgmodule)
-@PLATFORM_WIN32_TRUE@libadd_libglib = $(libglib)
+libadd_libgmodule = $(libgmodule)
+libadd_libglib = $(libglib)
 @PLATFORM_WIN32_TRUE@no_undefined = -no-undefined
 
 @PLATFORM_WIN32_TRUE@module_test_exp = module-test.exp
