--- glib-2.12.6/glib/gkeyfile.c~	2006-12-19 22:03:00.000000000 +0100
+++ glib-2.12.6/glib/gkeyfile.c	2006-12-29 03:08:38.000000000 +0100
@@ -3238,7 +3238,7 @@ g_key_file_is_key_name (const gchar *nam
    * since gnome-vfs uses mime-types as keys in its cache.
    */
   while (*q && (g_unichar_isalnum (g_utf8_get_char (q)) || 
-                *q == '-' || *q == '_' || *q == '/' || *q == '+' || *q == '.'))
+                *q == '-' || *q == '_' || *q == '/' || *q == '+' || *q == '.' || *q == ' '))
     q = g_utf8_next_char (q);
   
   if (*q == '[')
