diff -urN glib-2.12.4.orig/m4macros/glib-gettext.m4 glib-2.12.4/m4macros/glib-gettext.m4
--- glib-2.12.4.orig/m4macros/glib-gettext.m4	2005-03-11 19:02:35.000000000 +0100
+++ glib-2.12.4/m4macros/glib-gettext.m4	2006-11-16 10:20:59.000000000 +0100
@@ -376,8 +376,10 @@
 [glib_REQUIRE([GLIB_GNU_GETTEXT])dnl
 glib_save_prefix="$prefix"
 glib_save_exec_prefix="$exec_prefix"
+glib_save_datarootdir="$datarootdir"
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
 test "x$exec_prefix" = xNONE && exec_prefix=$prefix
+datarootdir=`eval echo "${datarootdir}"`
 if test "x$CATOBJEXT" = "x.mo" ; then
   localedir=`eval echo "${libdir}/locale"`
 else
@@ -385,6 +387,7 @@
 fi
 prefix="$glib_save_prefix"
 exec_prefix="$glib_save_exec_prefix"
+datarootdir="$glib_save_datarootdir"
 AC_DEFINE_UNQUOTED($1, "$localedir",
   [Define the location where the catalogs will be installed])
 ])
