--- glib2.0-2.18.0.orig/gio/gdesktopappinfo.c	2008-09-02 17:09:19.000000000 +0200
+++ glib2.0-2.18.0/gio/gdesktopappinfo.c	2008-09-03 00:53:23.000000000 +0200
@@ -112,11 +112,12 @@
   data_dirs = g_get_system_data_dirs ();
   length = g_strv_length ((char **) data_dirs);
   
-  args = g_new (char *, length + 2);
+  args = g_new (char *, length + 3);
   
   j = 0;
   user_data_dir = g_get_user_data_dir ();
   args[j++] = g_build_filename (user_data_dir, "applications", NULL);
+  args[j++] = g_strdup("/usr/share/gnome/applications");
   for (i = 0; i < length; i++)
     args[j++] = g_build_filename (data_dirs[i],
 				  "applications", NULL);
