Index: glib2.0-2.25.10/gio/gunixmounts.c
===================================================================
--- glib2.0-2.25.10.orig/gio/gunixmounts.c	2010-06-20 19:59:40.000000000 +0200
+++ glib2.0-2.25.10/gio/gunixmounts.c	2010-06-24 20:01:24.869942012 +0200
@@ -228,6 +228,8 @@
     "/home",
     "/lib",
     "/lib64",
+    "/live/cow",
+    "/live/image",
     "/media",
     "/mnt",
     "/opt",
@@ -238,7 +240,12 @@
     "/usr",
     "/usr/local",
     "/var",
+    "/var/crash",
+    "/var/local",
+    "/var/log",
     "/var/log/audit", /* https://bugzilla.redhat.com/show_bug.cgi?id=333041 */
+    "/var/mail",
+    "/var/run",
     "/var/tmp",       /* https://bugzilla.redhat.com/show_bug.cgi?id=335241 */
     "/proc",
     "/sbin",
