--- a/gio/gunixmounts.c	2008-09-12 19:55:52.000000000 +0200
+++ b/gio/gunixmounts.c	2008-09-26 16:12:33.000000000 +0200
@@ -221,6 +221,8 @@
     "/home",
     "/lib",
     "/lib64",
+    "/live/cow",
+    "/live/image",
     "/media",
     "/mnt",
     "/opt",
@@ -231,7 +233,12 @@
     "/usr",
     "/usr/local",
     "/var",
+    "/var/crash",
+    "/var/local",
+    "/var/log",
     "/var/log/audit", /* https://bugzilla.redhat.com/show_bug.cgi?id=333041 */
+    "/var/mail",
+    "/var/run",
     "/var/tmp",       /* https://bugzilla.redhat.com/show_bug.cgi?id=335241 */
     "/proc",
     "/sbin",
