--- glib-2.16.4/configure.in.old	2008-07-14 14:39:51.000000000 +0200
+++ glib-2.16.4/configure.in	2008-07-14 14:40:33.000000000 +0200
@@ -831,6 +831,19 @@
 
 # check for bytesex stuff
 AC_C_BIGENDIAN
+if test x$ac_cv_c_bigendian = xuniversal ; then
+AC_TRY_COMPILE([#include <endian.h>], [#if __BYTE_ORDER == __BIG_ENDIAN
+#else
+#error Not a big endian. 
+#endif],
+    ac_cv_c_bigendian=yes
+    ,AC_TRY_COMPILE([#include <endian.h>], [#if __BYTE_ORDER == __LITTLE_ENDIAN
+#else
+#error Not a little endian. 
+#endif],
+    ac_cv_c_bigendian=no
+    ,AC_MSG_WARN([Could not determine endianness.])))
+fi
 
 # check for header files
 AC_CHECK_HEADERS([dirent.h float.h limits.h pwd.h grp.h sys/param.h sys/poll.h sys/resource.h])
@@ -3137,18 +3150,15 @@
 g_have_gnuc_visibility=$g_have_gnuc_visibility
 g_have_sunstudio_visibility=$g_have_sunstudio_visibility
 
-case xyes in
-x$ac_cv_c_bigendian)
+if test x$ac_cv_c_bigendian = xyes; then
   g_byte_order=G_BIG_ENDIAN
   g_bs_native=BE
   g_bs_alien=LE
-  ;;
-*)
+else
   g_byte_order=G_LITTLE_ENDIAN
   g_bs_native=LE
   g_bs_alien=BE
-  ;;
-esac
+fi
 
 g_pollin=$glib_cv_value_POLLIN
 g_pollout=$glib_cv_value_POLLOUT
