Index: glib2.0-2.19.6/gio/gfileattribute-priv.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gfileattribute-priv.h	2009-02-15 19:10:34.000000000 -0300
+++ glib2.0-2.19.6/gio/gfileattribute-priv.h	2009-02-15 19:39:35.000000000 -0300
@@ -42,48 +42,48 @@
   } u;
 } GFileAttributeValue;
 
-GFileAttributeValue *_g_file_attribute_value_new             (void);
-void                 _g_file_attribute_value_free            (GFileAttributeValue *attr);
-void                 _g_file_attribute_value_clear           (GFileAttributeValue *attr);
-void                 _g_file_attribute_value_set             (GFileAttributeValue *attr,
-							      const GFileAttributeValue *new_value);
-GFileAttributeValue *_g_file_attribute_value_dup             (const GFileAttributeValue *other);
-gpointer             _g_file_attribute_value_peek_as_pointer (GFileAttributeValue *attr);
-
-char *               _g_file_attribute_value_as_string       (const GFileAttributeValue *attr);
-
-const char *         _g_file_attribute_value_get_string      (const GFileAttributeValue *attr);
-const char *         _g_file_attribute_value_get_byte_string (const GFileAttributeValue *attr);
-gboolean             _g_file_attribute_value_get_boolean     (const GFileAttributeValue *attr);
-guint32              _g_file_attribute_value_get_uint32      (const GFileAttributeValue *attr);
-gint32               _g_file_attribute_value_get_int32       (const GFileAttributeValue *attr);
-guint64              _g_file_attribute_value_get_uint64      (const GFileAttributeValue *attr);
-gint64               _g_file_attribute_value_get_int64       (const GFileAttributeValue *attr);
-GObject *            _g_file_attribute_value_get_object      (const GFileAttributeValue *attr);
-
-void                 _g_file_attribute_value_set_from_pointer(GFileAttributeValue *attr,
-							      GFileAttributeType   type,
-							      gpointer             value_p,
-							      gboolean             dup);
-void                 _g_file_attribute_value_set_string      (GFileAttributeValue *attr,
-							      const char          *string);
-void                 _g_file_attribute_value_set_byte_string (GFileAttributeValue *attr,
-							      const char          *string);
-void                 _g_file_attribute_value_set_boolean     (GFileAttributeValue *attr,
-							      gboolean             value);
-void                 _g_file_attribute_value_set_uint32      (GFileAttributeValue *attr,
-							      guint32              value);
-void                 _g_file_attribute_value_set_int32       (GFileAttributeValue *attr,
-							      gint32               value);
-void                 _g_file_attribute_value_set_uint64      (GFileAttributeValue *attr,
-							      guint64              value);
-void                 _g_file_attribute_value_set_int64       (GFileAttributeValue *attr,
-							      gint64               value);
-void                 _g_file_attribute_value_set_object      (GFileAttributeValue *attr,
-							      GObject             *obj);
+G_GNUC_INTERNAL GFileAttributeValue *_g_file_attribute_value_new             (void);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_free            (GFileAttributeValue *attr);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_clear           (GFileAttributeValue *attr);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set             (GFileAttributeValue *attr,
+		                					      const GFileAttributeValue *new_value);
+G_GNUC_INTERNAL GFileAttributeValue *_g_file_attribute_value_dup             (const GFileAttributeValue *other);
+G_GNUC_INTERNAL gpointer             _g_file_attribute_value_peek_as_pointer (GFileAttributeValue *attr);
+
+G_GNUC_INTERNAL char *               _g_file_attribute_value_as_string       (const GFileAttributeValue *attr);
+
+G_GNUC_INTERNAL const char *         _g_file_attribute_value_get_string      (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL const char *         _g_file_attribute_value_get_byte_string (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL gboolean             _g_file_attribute_value_get_boolean     (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL guint32              _g_file_attribute_value_get_uint32      (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL gint32               _g_file_attribute_value_get_int32       (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL guint64              _g_file_attribute_value_get_uint64      (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL gint64               _g_file_attribute_value_get_int64       (const GFileAttributeValue *attr);
+G_GNUC_INTERNAL GObject *            _g_file_attribute_value_get_object      (const GFileAttributeValue *attr);
+
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_from_pointer(GFileAttributeValue *attr,
+		                  					      GFileAttributeType   type,
+				                     			      gpointer             value_p,
+						                	      gboolean             dup);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_string      (GFileAttributeValue *attr,
+                    							      const char          *string);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_byte_string (GFileAttributeValue *attr,
+		                    					      const char          *string);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_boolean     (GFileAttributeValue *attr,
+				                			      gboolean             value);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_uint32      (GFileAttributeValue *attr,
+                     							      guint32              value);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_int32       (GFileAttributeValue *attr,
+		                   					      gint32               value);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_uint64      (GFileAttributeValue *attr,
+                                                                              guint64              value);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_int64       (GFileAttributeValue *attr,
+                                                                              gint64               value);
+G_GNUC_INTERNAL void                 _g_file_attribute_value_set_object      (GFileAttributeValue *attr,
+                                                                              GObject             *obj);
 
 
-GFileAttributeValue *_g_file_info_get_attribute_value (GFileInfo  *info,
-						       const char *attribute);
+G_GNUC_INTERNAL GFileAttributeValue *_g_file_info_get_attribute_value (GFileInfo  *info,
+                                                                       const char *attribute);
 
 #endif /* __G_FILE_ATTRIBUTE_PRIV_H__ */
Index: glib2.0-2.19.6/gio/gasynchelper.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gasynchelper.h	2009-02-15 19:42:28.000000000 -0300
+++ glib2.0-2.19.6/gio/gasynchelper.h	2009-02-15 19:42:36.000000000 -0300
@@ -38,15 +38,15 @@
 				   GIOCondition condition,
 				   int          fd);
 
-void     _g_queue_async_result (GAsyncResultData *result,
-				gpointer         async_object,
-				GError          *error,
-				gpointer         user_data,
-				GSourceFunc      source_func);
+G_GNUC_INTERNAL void     _g_queue_async_result (GAsyncResultData *result,
+                                                gpointer         async_object,
+                                                GError          *error,
+                                                gpointer         user_data,
+                                                GSourceFunc      source_func);
 
-GSource *_g_fd_source_new      (int              fd,
-				gushort          events,
-				GCancellable    *cancellable);
+G_GNUC_INTERNAL GSource *_g_fd_source_new      (int              fd,
+                                                gushort          events,
+                                                GCancellable    *cancellable);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gdummyfile.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gdummyfile.h	2009-02-15 19:43:33.000000000 -0300
+++ glib2.0-2.19.6/gio/gdummyfile.h	2009-02-15 19:44:04.000000000 -0300
@@ -42,9 +42,9 @@
   GObjectClass parent_class;
 };
 
-GType   _g_dummy_file_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType   _g_dummy_file_get_type (void) G_GNUC_CONST;
 
-GFile * _g_dummy_file_new      (const char *uri);
+G_GNUC_INTERNAL GFile * _g_dummy_file_new      (const char *uri);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gcontenttypeprivate.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gcontenttypeprivate.h	2009-02-15 20:04:22.000000000 -0300
+++ glib2.0-2.19.6/gio/gcontenttypeprivate.h	2009-02-15 20:04:44.000000000 -0300
@@ -27,9 +27,9 @@
 
 G_BEGIN_DECLS
 
-gsize  _g_unix_content_type_get_sniff_len (void);
-char * _g_unix_content_type_unalias       (const char *type);
-char **_g_unix_content_type_get_parents   (const char *type);
+G_GNUC_INTERNAL gsize  _g_unix_content_type_get_sniff_len (void);
+G_GNUC_INTERNAL char * _g_unix_content_type_unalias       (const char *type);
+G_GNUC_INTERNAL char **_g_unix_content_type_get_parents   (const char *type);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/giomodule-priv.h
===================================================================
--- glib2.0-2.19.6.orig/gio/giomodule-priv.h	2009-02-15 19:47:03.000000000 -0300
+++ glib2.0-2.19.6/gio/giomodule-priv.h	2009-02-15 19:47:07.000000000 -0300
@@ -27,7 +27,7 @@
 
 G_BEGIN_DECLS
 
-void       _g_io_modules_ensure_loaded (void);
+G_GNUC_INTERNAL void       _g_io_modules_ensure_loaded (void);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocaldirectorymonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocaldirectorymonitor.h	2009-02-15 19:48:21.000000000 -0300
+++ glib2.0-2.19.6/gio/glocaldirectorymonitor.h	2009-02-15 20:44:16.000000000 -0300
@@ -61,9 +61,9 @@
 
 GType           g_local_directory_monitor_get_type (void) G_GNUC_CONST;
 
-GFileMonitor * _g_local_directory_monitor_new      (const char         *dirname,
-                                                    GFileMonitorFlags   flags,
-                                                    GError            **error);
+G_GNUC_INTERNAL GFileMonitor * _g_local_directory_monitor_new      (const char         *dirname,
+                                                                    GFileMonitorFlags   flags,
+                                                                    GError            **error);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalfile.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalfile.h	2009-02-15 19:51:17.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalfile.h	2009-02-15 19:51:24.000000000 -0300
@@ -42,9 +42,9 @@
   GObjectClass parent_class;
 };
 
-GType   _g_local_file_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType   _g_local_file_get_type (void) G_GNUC_CONST;
 
-GFile * _g_local_file_new      (const char *filename);
+G_GNUC_INTERNAL GFile * _g_local_file_new      (const char *filename);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalfileenumerator.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalfileenumerator.h	2009-02-15 19:49:16.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalfileenumerator.h	2009-02-15 19:49:30.000000000 -0300
@@ -44,13 +44,13 @@
   GFileEnumeratorClass parent_class;
 };
 
-GType             _g_local_file_enumerator_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType             _g_local_file_enumerator_get_type (void) G_GNUC_CONST;
 
-GFileEnumerator * _g_local_file_enumerator_new      (GLocalFile           *file,
-                                                     const char           *attributes,
-                                                     GFileQueryInfoFlags   flags,
-                                                     GCancellable         *cancellable,
-                                                     GError              **error);
+G_GNUC_INTERNAL GFileEnumerator * _g_local_file_enumerator_new      (GLocalFile           *file,
+                                                                     const char           *attributes,
+                                                                     GFileQueryInfoFlags   flags,
+                                                                     GCancellable         *cancellable,
+                                                                     GError              **error);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalfileinfo.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalfileinfo.h	2009-02-15 19:51:31.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalfileinfo.h	2009-02-15 19:51:54.000000000 -0300
@@ -48,33 +48,33 @@
 #define GLocalFileStat struct stat
 #endif
 
-gboolean   _g_local_file_has_trash_dir        (const char             *dirname,
-                                               dev_t                   dir_dev);
-void       _g_local_file_info_get_parent_info (const char             *dir,
-                                               GFileAttributeMatcher  *attribute_matcher,
-                                               GLocalParentFileInfo   *parent_info);
-GFileInfo *_g_local_file_info_get             (const char             *basename,
-                                               const char             *path,
-                                               GFileAttributeMatcher  *attribute_matcher,
-                                               GFileQueryInfoFlags     flags,
-                                               GLocalParentFileInfo   *parent_info,
-                                               GError                **error);
-GFileInfo *_g_local_file_info_get_from_fd     (int                     fd,
-                                               char                   *attributes,
-                                               GError                **error);
-char *     _g_local_file_info_create_etag     (GLocalFileStat         *statbuf);
-gboolean   _g_local_file_info_set_attribute   (char                   *filename,
-                                               const char             *attribute,
-                                               GFileAttributeType      type,
-                                               gpointer                value_p,
-                                               GFileQueryInfoFlags     flags,
-                                               GCancellable           *cancellable,
-                                               GError                **error);
-gboolean   _g_local_file_info_set_attributes  (char                   *filename,
-                                               GFileInfo              *info,
-                                               GFileQueryInfoFlags     flags,
-                                               GCancellable           *cancellable,
-                                               GError                **error);
+G_GNUC_INTERNAL gboolean   _g_local_file_has_trash_dir        (const char             *dirname,
+                                                               dev_t                   dir_dev);
+G_GNUC_INTERNAL void       _g_local_file_info_get_parent_info (const char             *dir,
+                                                               GFileAttributeMatcher  *attribute_matcher,
+                                                               GLocalParentFileInfo   *parent_info);
+G_GNUC_INTERNAL GFileInfo *_g_local_file_info_get             (const char             *basename,
+                                                               const char             *path,
+                                                               GFileAttributeMatcher  *attribute_matcher,
+                                                               GFileQueryInfoFlags     flags,
+                                                               GLocalParentFileInfo   *parent_info,
+                                                               GError                **error);
+G_GNUC_INTERNAL GFileInfo *_g_local_file_info_get_from_fd     (int                     fd,
+                                                               char                   *attributes,
+                                                               GError                **error);
+G_GNUC_INTERNAL char *     _g_local_file_info_create_etag     (GLocalFileStat         *statbuf);
+G_GNUC_INTERNAL gboolean   _g_local_file_info_set_attribute   (char                   *filename,
+                                                               const char             *attribute,
+                                                               GFileAttributeType      type,
+                                                               gpointer                value_p,
+                                                               GFileQueryInfoFlags     flags,
+                                                               GCancellable           *cancellable,
+                                                               GError                **error);
+G_GNUC_INTERNAL gboolean   _g_local_file_info_set_attributes  (char                   *filename,
+                                                               GFileInfo              *info,
+                                                               GFileQueryInfoFlags     flags,
+                                                               GCancellable           *cancellable,
+                                                               GError                **error);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalfileinputstream.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalfileinputstream.h	2009-02-15 19:52:31.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalfileinputstream.h	2009-02-15 19:52:36.000000000 -0300
@@ -51,9 +51,9 @@
   GFileInputStreamClass parent_class;
 };
 
-GType              _g_local_file_input_stream_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType              _g_local_file_input_stream_get_type (void) G_GNUC_CONST;
 
-GFileInputStream * _g_local_file_input_stream_new      (int fd);
+G_GNUC_INTERNAL GFileInputStream * _g_local_file_input_stream_new      (int fd);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalfilemonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalfilemonitor.h	2009-02-15 19:53:57.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalfilemonitor.h	2009-02-15 20:44:09.000000000 -0300
@@ -54,9 +54,9 @@
 
 GType           g_local_file_monitor_get_type (void) G_GNUC_CONST;
 
-GFileMonitor * _g_local_file_monitor_new      (const char         *pathname,
-                                               GFileMonitorFlags   flags,
-                                               GError            **error);
+G_GNUC_INTERNAL GFileMonitor * _g_local_file_monitor_new      (const char         *pathname,
+                                                               GFileMonitorFlags   flags,
+                                                               GError            **error);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalfileoutputstream.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalfileoutputstream.h	2009-02-15 19:53:15.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalfileoutputstream.h	2009-02-15 19:53:23.000000000 -0300
@@ -51,21 +51,21 @@
   GFileOutputStreamClass parent_class;
 };
 
-GType               _g_local_file_output_stream_get_type (void) G_GNUC_CONST;
-GFileOutputStream * _g_local_file_output_stream_create   (const char       *filename,
-                                                          GFileCreateFlags  flags,
-                                                          GCancellable     *cancellable,
-                                                          GError          **error);
-GFileOutputStream * _g_local_file_output_stream_append   (const char       *filename,
-                                                          GFileCreateFlags  flags,
-                                                          GCancellable     *cancellable,
-                                                          GError          **error);
-GFileOutputStream * _g_local_file_output_stream_replace  (const char       *filename,
-                                                          const char       *etag,
-                                                          gboolean          create_backup,
-                                                          GFileCreateFlags  flags,
-                                                          GCancellable     *cancellable,
-                                                          GError          **error);
+G_GNUC_INTERNAL GType               _g_local_file_output_stream_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GFileOutputStream * _g_local_file_output_stream_create   (const char       *filename,
+                                                                          GFileCreateFlags  flags,
+                                                                          GCancellable     *cancellable,
+                                                                          GError          **error);
+G_GNUC_INTERNAL GFileOutputStream * _g_local_file_output_stream_append   (const char       *filename,
+                                                                          GFileCreateFlags  flags,
+                                                                          GCancellable     *cancellable,
+                                                                          GError          **error);
+G_GNUC_INTERNAL GFileOutputStream * _g_local_file_output_stream_replace  (const char       *filename,
+                                                                          const char       *etag,
+                                                                          gboolean          create_backup,
+                                                                          GFileCreateFlags  flags,
+                                                                          GCancellable     *cancellable,
+                                                                          GError          **error);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/glocalvfs.h
===================================================================
--- glib2.0-2.19.6.orig/gio/glocalvfs.h	2009-02-15 19:58:13.000000000 -0300
+++ glib2.0-2.19.6/gio/glocalvfs.h	2009-02-15 19:58:21.000000000 -0300
@@ -37,9 +37,9 @@
 typedef struct _GLocalVfs       GLocalVfs;
 typedef struct _GLocalVfsClass  GLocalVfsClass;
 
-GType   _g_local_vfs_get_type  (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType   _g_local_vfs_get_type  (void) G_GNUC_CONST;
 
-GVfs  * _g_local_vfs_new       (void);
+G_GNUC_INTERNAL GVfs  * _g_local_vfs_new       (void);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gmountprivate.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gmountprivate.h	2009-02-15 20:00:24.000000000 -0300
+++ glib2.0-2.19.6/gio/gmountprivate.h	2009-02-15 20:00:30.000000000 -0300
@@ -27,8 +27,8 @@
 
 G_BEGIN_DECLS
 
-GMount *_g_mount_get_for_mount_path (const char *mount_path,
-				     GCancellable *cancellable);
+G_GNUC_INTERNAL GMount *_g_mount_get_for_mount_path (const char *mount_path,
+                                                     GCancellable *cancellable);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gpollfilemonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gpollfilemonitor.h	2009-02-15 20:02:42.000000000 -0300
+++ glib2.0-2.19.6/gio/gpollfilemonitor.h	2009-02-15 20:02:59.000000000 -0300
@@ -41,9 +41,9 @@
   GFileMonitorClass parent_class;
 };
 
-GType          _g_poll_file_monitor_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType          _g_poll_file_monitor_get_type (void) G_GNUC_CONST;
 
-GFileMonitor * _g_poll_file_monitor_new      (GFile *file);
+G_GNUC_INTERNAL GFileMonitor * _g_poll_file_monitor_new      (GFile *file);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gunionvolumemonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gunionvolumemonitor.h	2009-02-15 20:03:33.000000000 -0300
+++ glib2.0-2.19.6/gio/gunionvolumemonitor.h	2009-02-15 20:03:48.000000000 -0300
@@ -42,7 +42,7 @@
   GVolumeMonitorClass parent_class;
 };
 
-GType _g_union_volume_monitor_get_type (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType _g_union_volume_monitor_get_type (void) G_GNUC_CONST;
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gunixmount.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gunixmount.h	2009-02-15 20:10:30.000000000 -0300
+++ glib2.0-2.19.6/gio/gunixmount.h	2009-02-15 20:10:49.000000000 -0300
@@ -43,16 +43,16 @@
   GObjectClass parent_class;
 };
 
-GType        _g_unix_mount_get_type       (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType        _g_unix_mount_get_type       (void) G_GNUC_CONST;
 
-GUnixMount * _g_unix_mount_new            (GVolumeMonitor  *volume_monitor,
-                                           GUnixMountEntry *mount_entry,
-                                           GUnixVolume     *volume);
-gboolean     _g_unix_mount_has_mount_path (GUnixMount      *mount,
-                                           const char      *mount_path);
-void         _g_unix_mount_unset_volume   (GUnixMount      *mount,
-                                           GUnixVolume     *volume);
-void         _g_unix_mount_unmounted      (GUnixMount      *mount);
+G_GNUC_INTERNAL GUnixMount * _g_unix_mount_new            (GVolumeMonitor  *volume_monitor,
+                                                           GUnixMountEntry *mount_entry,
+                                                           GUnixVolume     *volume);
+G_GNUC_INTERNAL gboolean     _g_unix_mount_has_mount_path (GUnixMount      *mount,
+                                                           const char      *mount_path);
+G_GNUC_INTERNAL void         _g_unix_mount_unset_volume   (GUnixMount      *mount,
+                                                           GUnixVolume     *volume);
+G_GNUC_INTERNAL void         _g_unix_mount_unmounted      (GUnixMount      *mount);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gunixvolume.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gunixvolume.h	2009-02-15 20:11:30.000000000 -0300
+++ glib2.0-2.19.6/gio/gunixvolume.h	2009-02-15 20:12:14.000000000 -0300
@@ -43,17 +43,17 @@
   GObjectClass parent_class;
 };
 
-GType         _g_unix_volume_get_type       (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType         _g_unix_volume_get_type       (void) G_GNUC_CONST;
 
-GUnixVolume * _g_unix_volume_new            (GVolumeMonitor  *volume_monitor,
-                                             GUnixMountPoint *mountpoint);
-gboolean      _g_unix_volume_has_mount_path (GUnixVolume     *volume,
-                                             const char      *mount_path);
-void          _g_unix_volume_set_mount      (GUnixVolume     *volume,
-                                             GUnixMount      *mount);
-void          _g_unix_volume_unset_mount    (GUnixVolume     *volume,
-                                             GUnixMount      *mount);
-void          _g_unix_volume_disconnected   (GUnixVolume     *volume);
+G_GNUC_INTERNAL GUnixVolume * _g_unix_volume_new            (GVolumeMonitor  *volume_monitor,
+                                                             GUnixMountPoint *mountpoint);
+G_GNUC_INTERNAL gboolean      _g_unix_volume_has_mount_path (GUnixVolume     *volume,
+                                                             const char      *mount_path);
+G_GNUC_INTERNAL void          _g_unix_volume_set_mount      (GUnixVolume     *volume,
+                                                             GUnixMount      *mount);
+G_GNUC_INTERNAL void          _g_unix_volume_unset_mount    (GUnixVolume     *volume,
+                                                             GUnixMount      *mount);
+G_GNUC_INTERNAL void          _g_unix_volume_disconnected   (GUnixVolume     *volume);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/gunixvolumemonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/gunixvolumemonitor.h	2009-02-15 20:12:33.000000000 -0300
+++ glib2.0-2.19.6/gio/gunixvolumemonitor.h	2009-02-15 20:12:46.000000000 -0300
@@ -52,11 +52,11 @@
   GNativeVolumeMonitorClass parent_class;
 };
 
-GType            _g_unix_volume_monitor_get_type                     (void) G_GNUC_CONST;
+G_GNUC_INTERNAL GType            _g_unix_volume_monitor_get_type                     (void) G_GNUC_CONST;
 
-GVolumeMonitor * _g_unix_volume_monitor_new                          (void);
-GUnixVolume    * _g_unix_volume_monitor_lookup_volume_for_mount_path (GUnixVolumeMonitor *monitor,
-                                                                      const char         *mount_path);
+G_GNUC_INTERNAL GVolumeMonitor * _g_unix_volume_monitor_new                          (void);
+G_GNUC_INTERNAL GUnixVolume    * _g_unix_volume_monitor_lookup_volume_for_mount_path (GUnixVolumeMonitor *monitor,
+                                                                                      const char         *mount_path);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/inotify/ginotifydirectorymonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/ginotifydirectorymonitor.h	2009-02-15 19:45:36.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/ginotifydirectorymonitor.h	2009-02-15 19:45:44.000000000 -0300
@@ -46,7 +46,7 @@
   GLocalDirectoryMonitorClass parent_class;
 };
 
-GType _g_inotify_directory_monitor_get_type (void);
+G_GNUC_INTERNAL GType _g_inotify_directory_monitor_get_type (void);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/inotify/ginotifyfilemonitor.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/ginotifyfilemonitor.h	2009-02-15 19:46:03.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/ginotifyfilemonitor.h	2009-02-15 19:46:04.000000000 -0300
@@ -47,7 +47,7 @@
   GLocalFileMonitorClass parent_class;
 };
 
-GType _g_inotify_file_monitor_get_type (void);
+G_GNUC_INTERNAL GType _g_inotify_file_monitor_get_type (void);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gio/inotify/inotify-diag.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/inotify-diag.h	2009-02-15 20:14:33.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/inotify-diag.h	2009-02-15 20:14:37.000000000 -0300
@@ -24,6 +24,6 @@
 #ifndef __INOTIFY_DIAG_H
 #define __INOTIFY_DIAG_H
 
-void _id_startup (void);
+G_GNUC_INTERNAL void _id_startup (void);
 
 #endif /* __INOTIFY_DIAG_H */
Index: glib2.0-2.19.6/gio/inotify/inotify-helper.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/inotify-helper.h	2009-02-15 20:13:17.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/inotify-helper.h	2009-02-15 20:13:29.000000000 -0300
@@ -26,8 +26,8 @@
 
 #include "inotify-sub.h"
 
-gboolean _ih_startup    (void);
-gboolean _ih_sub_add    (inotify_sub *sub);
-gboolean _ih_sub_cancel (inotify_sub *sub);
+G_GNUC_INTERNAL gboolean _ih_startup    (void);
+G_GNUC_INTERNAL gboolean _ih_sub_add    (inotify_sub *sub);
+G_GNUC_INTERNAL gboolean _ih_sub_cancel (inotify_sub *sub);
 
 #endif /* __INOTIFY_HELPER_H */
Index: glib2.0-2.19.6/gio/inotify/inotify-kernel.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/inotify-kernel.h	2009-02-15 20:14:54.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/inotify-kernel.h	2009-02-15 20:15:16.000000000 -0300
@@ -32,24 +32,24 @@
   struct ik_event_s *pair;
 } ik_event_t;
 
-gboolean _ik_startup (void (*cb) (ik_event_t *event));
+G_GNUC_INTERNAL gboolean _ik_startup (void (*cb) (ik_event_t *event));
 
-ik_event_t *_ik_event_new_dummy (const char *name,
-				 gint32      wd,
-				 guint32     mask);
-void        _ik_event_free      (ik_event_t *event);
-
-gint32      _ik_watch           (const char *path,
-				 guint32     mask,
-				 int        *err);
-int         _ik_ignore          (const char *path,
-				 gint32      wd);
+G_GNUC_INTERNAL ik_event_t *_ik_event_new_dummy (const char *name,
+                                                 gint32      wd,
+                                                 guint32     mask);
+G_GNUC_INTERNAL void        _ik_event_free      (ik_event_t *event);
+
+G_GNUC_INTERNAL gint32      _ik_watch           (const char *path,
+                                                 guint32     mask,
+                                                 int        *err);
+G_GNUC_INTERNAL int         _ik_ignore          (const char *path,
+                                                 gint32      wd);
 
 
 /* The miss count will probably be enflated */
-void        _ik_move_stats     (guint32 *matches,
-				guint32 *misses);
-const char *_ik_mask_to_string (guint32  mask);
+G_GNUC_INTERNAL void        _ik_move_stats     (guint32 *matches,
+                                                guint32 *misses);
+G_GNUC_INTERNAL const char *_ik_mask_to_string (guint32  mask);
 
 
 #endif
Index: glib2.0-2.19.6/gio/inotify/inotify-missing.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/inotify-missing.h	2009-02-15 20:15:28.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/inotify-missing.h	2009-02-15 20:15:34.000000000 -0300
@@ -26,10 +26,10 @@
 
 #include "inotify-sub.h"
 
-void _im_startup   (void (*missing_cb)(inotify_sub *sub));
-void _im_add       (inotify_sub *sub);
-void _im_rm        (inotify_sub *sub);
-void _im_diag_dump (GIOChannel  *ioc);
+G_GNUC_INTERNAL void _im_startup   (void (*missing_cb)(inotify_sub *sub));
+G_GNUC_INTERNAL void _im_add       (inotify_sub *sub);
+G_GNUC_INTERNAL void _im_rm        (inotify_sub *sub);
+G_GNUC_INTERNAL void _im_diag_dump (GIOChannel  *ioc);
 
 
 #endif /* __INOTIFY_MISSING_H */
Index: glib2.0-2.19.6/gio/inotify/inotify-path.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/inotify-path.h	2009-02-15 20:15:44.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/inotify-path.h	2009-02-15 20:15:49.000000000 -0300
@@ -26,8 +26,8 @@
 #include "inotify-kernel.h"
 #include "inotify-sub.h"
 
-gboolean _ip_startup (void (*event_cb)(ik_event_t *event, inotify_sub *sub));
-gboolean _ip_start_watching (inotify_sub *sub);
-gboolean _ip_stop_watching  (inotify_sub *sub);
+G_GNUC_INTERNAL gboolean _ip_startup (void (*event_cb)(ik_event_t *event, inotify_sub *sub));
+G_GNUC_INTERNAL gboolean _ip_start_watching (inotify_sub *sub);
+G_GNUC_INTERNAL gboolean _ip_stop_watching  (inotify_sub *sub);
 
 #endif
Index: glib2.0-2.19.6/gio/inotify/inotify-sub.h
===================================================================
--- glib2.0-2.19.6.orig/gio/inotify/inotify-sub.h	2009-02-15 20:14:09.000000000 -0300
+++ glib2.0-2.19.6/gio/inotify/inotify-sub.h	2009-02-15 20:14:15.000000000 -0300
@@ -32,7 +32,7 @@
 	gpointer user_data;
 } inotify_sub;
 
-inotify_sub* _ih_sub_new (const gchar* dirname, const gchar* filename, gpointer user_data);
-void         _ih_sub_free (inotify_sub* sub);
+G_GNUC_INTERNAL inotify_sub* _ih_sub_new (const gchar* dirname, const gchar* filename, gpointer user_data);
+G_GNUC_INTERNAL void         _ih_sub_free (inotify_sub* sub);
 
 #endif /* __INOTIFY_SUB_H */
Index: glib2.0-2.19.6/glib/gasyncqueue.h
===================================================================
--- glib2.0-2.19.6.orig/glib/gasyncqueue.h	2009-02-15 21:04:14.000000000 -0300
+++ glib2.0-2.19.6/glib/gasyncqueue.h	2009-02-15 21:04:35.000000000 -0300
@@ -113,7 +113,7 @@
 						 gpointer          user_data);
 
 /* Private API */
-GMutex*      _g_async_queue_get_mutex           (GAsyncQueue      *queue);
+G_GNUC_INTERNAL GMutex*      _g_async_queue_get_mutex           (GAsyncQueue      *queue);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/glib/gunicode.h
===================================================================
--- glib2.0-2.19.6.orig/glib/gunicode.h	2009-02-15 21:10:04.000000000 -0300
+++ glib2.0-2.19.6/glib/gunicode.h	2009-02-15 21:10:21.000000000 -0300
@@ -397,7 +397,7 @@
 
 /* private */
 
-gchar *_g_utf8_make_valid (const gchar *name);
+G_GNUC_INTERNAL gchar *_g_utf8_make_valid (const gchar *name);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/glib/gutils.c
===================================================================
--- glib2.0-2.19.6.orig/glib/gutils.c	2009-02-15 21:06:09.000000000 -0300
+++ glib2.0-2.19.6/glib/gutils.c	2009-02-15 21:06:13.000000000 -0300
@@ -2959,7 +2959,7 @@
  *       but it is big, ugly, and complicated, so I'm reluctant
  *       to do so when this should handle 99% of the time...
  */
-GSList *
+G_GNUC_INTERNAL GSList *
 _g_compute_locale_variants (const gchar *locale)
 {
   GSList *retval = NULL;
Index: glib2.0-2.19.6/glib/gutils.h
===================================================================
--- glib2.0-2.19.6.orig/glib/gutils.h	2009-02-15 21:08:04.000000000 -0300
+++ glib2.0-2.19.6/glib/gutils.h	2009-02-15 21:08:27.000000000 -0300
@@ -254,8 +254,8 @@
 gchar**               g_listenv            (void);
 
 /* private */
-const gchar*	     _g_getenv_nomalloc	   (const gchar	*variable,
-					    gchar        buffer[1024]);
+G_GNUC_INTERNAL const gchar*	     _g_getenv_nomalloc	   (const gchar	*variable,
+                                                            gchar        buffer[1024]);
 
 /* we try to provide a useful equivalent for ATEXIT if it is
  * not defined, but use is actually abandoned. people should
Index: glib2.0-2.19.6/gobject/gsignal.h
--- glib2.0-2.19.6/gobject/gsignal.h	2009-02-02 17:17:46.000000000 -0200
+++ glib2.0-2.19.6/gobject/gsignal.h	2009-02-15 22:29:31.000000000 -0300
@@ -502,7 +502,7 @@
 
 /*< private >*/
 void	 g_signal_handlers_destroy	      (gpointer		  instance);
-void	 _g_signals_destroy		      (GType		  itype);
+G_GNUC_INTERNAL void	 _g_signals_destroy		      (GType		  itype);
 
 G_END_DECLS
 
Index: glib2.0-2.19.6/gobject/gtype.c
--- glib2.0-2.19.6/gobject/gtype.c	2009-02-02 17:17:46.000000000 -0200
+++ glib2.0-2.19.6/gobject/gtype.c	2009-02-15 22:30:52.000000000 -0300
@@ -357,7 +357,7 @@
 static GQuark          static_quark_type_flags = 0;
 static GQuark          static_quark_iface_holder = 0;
 static GQuark          static_quark_dependants_array = 0;
-GTypeDebugFlags	       _g_type_debug_flags = 0;
+G_GNUC_INTERNAL GTypeDebugFlags	       _g_type_debug_flags = 0;
 
 
 /* --- type nodes --- */
