Description: Fix path for /bin/true in Debian
Author: Emilio Pozuelo Monfort <pochu@debian.org>
Origin: vendor

--- a/gio/tests/desktop-app-info.c
+++ b/gio/tests/desktop-app-info.c
@@ -34,7 +34,7 @@
   GAppInfo *info;
 
   error = NULL;
-  info = g_app_info_create_from_commandline ("/usr/bin/true blah",
+  info = g_app_info_create_from_commandline ("/bin/true blah",
                                              name,
                                              G_APP_INFO_CREATE_NONE,
                                              &error);
