Index: glib2.0-2.33.12+really2.32.4/gio/glib-compile-resources.c
===================================================================
--- glib2.0-2.33.12+really2.32.4.orig/gio/glib-compile-resources.c	2012-07-14 22:33:11.000000000 +0200
+++ glib2.0-2.33.12+really2.32.4/gio/glib-compile-resources.c	2012-09-28 21:32:32.168548276 +0200
@@ -272,7 +272,6 @@ end_element (GMarkupParseContext  *conte
       if (state->preproc_options)
         {
           gchar **options;
-          gchar *stderr_child = NULL;
           guint i;
           gboolean xml_stripblanks = FALSE;
           gboolean to_pixdata = FALSE;
@@ -298,6 +297,7 @@ end_element (GMarkupParseContext  *conte
           if (xml_stripblanks && xmllint != NULL)
             {
               gchar *argv[8];
+              gchar *stderr_child = NULL;
               int status, fd, argc;
 
               tmp_file = g_strdup ("resource-XXXXXXXX");
@@ -336,6 +336,7 @@ end_element (GMarkupParseContext  *conte
                 {
                   g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
                                _("Error processing input file with xmllint:\n%s"), stderr_child);
+                  g_free (stderr_child);
                   goto cleanup;
                 }
 #endif
@@ -392,6 +393,7 @@ end_element (GMarkupParseContext  *conte
                 {
                   g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
 			       _("Error processing input file with to-pixdata:\n%s"), stderr_child);
+                  g_free (stderr_child);
                   goto cleanup;
                 }
 #endif
