Index: glib2.0-2.27.1/configure
===================================================================
--- glib2.0-2.27.1.orig/configure	2010-10-29 11:29:10.368045795 +0200
+++ glib2.0-2.27.1/configure	2010-10-29 11:30:37.700013079 +0200
@@ -27049,17 +27042,18 @@
       if test "x$enable_dtrace" = xyes; then
         as_fn_error $? "dtrace not found" "$LINENO" 5
       fi
-    fi
-    ac_fn_c_check_header_mongrel "$LINENO" "sys/sdt.h" "ac_cv_header_sys_sdt_h" "$ac_includes_default"
+    else
+      ac_fn_c_check_header_mongrel "$LINENO" "sys/sdt.h" "ac_cv_header_sys_sdt_h" "$ac_includes_default"
 if test "x$ac_cv_header_sys_sdt_h" = x""yes; then :
   have_dtrace=yes
 else
   if test "x$enable_dtrace" = xyes; then
-                      as_fn_error $? "dtrace support needs sys/sdt.h header" "$LINENO" 5
-                     fi
+                        as_fn_error $? "dtrace support needs sys/sdt.h header" "$LINENO" 5
+                       fi
 fi
 
 
+    fi
   fi
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
Index: glib2.0-2.27.1/configure.ac
===================================================================
--- glib2.0-2.27.1.orig/configure.ac	2010-10-29 11:29:10.397028810 +0200
+++ glib2.0-2.27.1/configure.ac	2010-10-29 11:29:21.888024724 +0200
@@ -2785,11 +2785,12 @@
       if test "x$enable_dtrace" = xyes; then
         AC_MSG_ERROR([dtrace not found])
       fi
+    else
+      AC_CHECK_HEADER([sys/sdt.h],have_dtrace=yes,
+                      [if test "x$enable_dtrace" = xyes; then
+                        AC_MSG_ERROR([dtrace support needs sys/sdt.h header])
+                       fi])
     fi
-    AC_CHECK_HEADER([sys/sdt.h],have_dtrace=yes,
-                    [if test "x$enable_dtrace" = xyes; then
-                      AC_MSG_ERROR([dtrace support needs sys/sdt.h header])
-                     fi])
   fi
 else
   AC_MSG_RESULT([no])
