--- glib2.0-2.16.2/glib/gmain.c.orig	2008-03-31 22:45:04.000000000 +0200
+++ glib2.0-2.16.2/glib/gmain.c	2008-04-04 12:01:19.000000000 +0200
@@ -3320,7 +3320,7 @@
       if (!session_bus_address)
         session_bus_address = g_getenv ("HOSTNAME");
       if (session_bus_address)
-        timer_perturb = g_str_hash (session_bus_address);
+        timer_perturb = ABS ((gint) g_str_hash (session_bus_address));
       else
         timer_perturb = 0;
     }
