From: Emilio Pozuelo Monfort <pochu@debian.org>
Date: Sun, 9 Mar 2014 15:06:31 +0100
Subject: Skip the monitor test on non-linux as it currently hangs

Forwarded: no
---
 gio/tests/monitor.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/tests/monitor.c b/gio/tests/monitor.c
index 4d64fa8..01d91a0 100644
--- a/gio/tests/monitor.c
+++ b/gio/tests/monitor.c
@@ -214,7 +214,9 @@ main (int argc, char *argv[])
 {
   g_test_init (&argc, &argv, NULL);
 
+#ifdef __linux__
   g_test_add ("/monitor/directory", Fixture, NULL, setup, test_directory_monitor, teardown);
+#endif
 
   return g_test_run ();
 }
