From: Emilio Pozuelo Monfort <pochu@debian.org>
Date: Sun, 09 Mar 2014 15:06:31 +0100
Subject: Skip the monitor test on non-linux as it currently hangs

Forwarded: no

--- a/gio/tests/monitor.c
+++ b/gio/tests/monitor.c
@@ -178,7 +178,9 @@
 {
   g_test_init (&argc, &argv, NULL);
 
+#ifdef __linux__
   g_test_add_func ("/monitor/directory", test_directory_monitor);
+#endif
 
   return g_test_run ();
 }
