From: Steve Langasek <steve.langasek@canonical.com>
Subject: gio: add fallback directory for pre-multiarch compatibility
Date: Sat, 16 Jul 2011 13:12:48 -0700

Origin: vendor, Ubuntu
Bug-Debian: https://bugs.debian.org/634099
Forwarded: no, specific to Debian and its derivatives

Index: b/gio/giomodule.c
===================================================================
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1093,6 +1093,9 @@
 
       g_io_module_scope_free (scope);
 
+      /* Hard-coded fallback directory for pre-multiarch compatibility */
+      g_io_modules_scan_all_in_directory ("/usr/lib/gio/modules");
+
       /* Initialize types from built-in "modules" */
       g_type_ensure (g_null_settings_backend_get_type ());
       g_type_ensure (g_memory_settings_backend_get_type ());
