Index: glib2.0-exp/gio/tests/appinfo-test.desktop
===================================================================
--- glib2.0-exp.orig/gio/tests/appinfo-test.desktop	2013-01-15 11:12:11.007356023 +0100
+++ glib2.0-exp/gio/tests/appinfo-test.desktop	2013-01-15 11:12:11.003356023 +0100
@@ -9,7 +9,6 @@
 Comment[de]=GAppInfo Beispiel
 Exec=./appinfo-test --option %U %i --name %c --filename %k %m %%
 Icon=testicon.svg
-Terminal=true
 StartupNotify=true
 StartupWMClass=appinfo-class
 MimeType=image/png;image/jpeg;
Index: glib2.0-exp/gio/tests/desktop-app-info.c
===================================================================
--- glib2.0-exp.orig/gio/tests/desktop-app-info.c	2013-01-15 11:12:11.007356023 +0100
+++ glib2.0-exp/gio/tests/desktop-app-info.c	2013-01-15 11:12:43.843357612 +0100
@@ -360,16 +360,13 @@
   appinfo = g_desktop_app_info_new_from_filename (SRCDIR "/appinfo-test.desktop");
   g_assert (appinfo != NULL);
 
-  g_assert (g_desktop_app_info_has_key (appinfo, "Terminal"));
+  g_assert (g_desktop_app_info_has_key (appinfo, "StartupNotify"));
   g_assert (!g_desktop_app_info_has_key (appinfo, "Bratwurst"));
 
   s = g_desktop_app_info_get_string (appinfo, "StartupWMClass");
   g_assert_cmpstr (s, ==, "appinfo-class");
   g_free (s);
 
-  b = g_desktop_app_info_get_boolean (appinfo, "Terminal");
-  g_assert (b);
-
   g_object_unref (appinfo);
 }
 
