From 8d2c14401db67f647bcbcab2d22c02a0728dfceb Mon Sep 17 00:00:00 2001
From: David Zeuthen <davidz@redhat.com>
Date: Wed, 5 Oct 2011 10:56:11 -0400
Subject: [PATCH] GDBus: disable two more GDBus tests using fork()

Signed-off-by: David Zeuthen <davidz@redhat.com>
---
 gio/tests/gdbus-proxy.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/gio/tests/gdbus-proxy.c b/gio/tests/gdbus-proxy.c
index 975a9ac..2e280fc 100644
--- a/gio/tests/gdbus-proxy.c
+++ b/gio/tests/gdbus-proxy.c
@@ -441,6 +441,7 @@ test_bogus_method_return (GDBusProxy *proxy)
   g_assert (result == NULL);
 }
 
+#if 0 /* Disabled: see https://bugzilla.gnome.org/show_bug.cgi?id=658999 */
 static void
 test_bogus_signal (GDBusProxy *proxy)
 {
@@ -516,6 +517,7 @@ test_bogus_property (GDBusProxy *proxy)
   _g_assert_signal_received (proxy, "g-properties-changed");
   g_dbus_proxy_set_interface_info (proxy, old_iface_info);
 }
+#endif /* Disabled: see https://bugzilla.gnome.org/show_bug.cgi?id=658999 */
 
 /* ---------------------------------------------------------------------------------------------------- */
 
@@ -565,8 +567,10 @@ test_expected_interface (GDBusProxy *proxy)
 
   /* And also where we deliberately set the expected interface definition incorrectly */
   test_bogus_method_return (proxy);
+  /* Disabled: see https://bugzilla.gnome.org/show_bug.cgi?id=658999
   test_bogus_signal (proxy);
   test_bogus_property (proxy);
+  */
 
   /* Also check that we complain if setting a cached property of the wrong type */
   if (g_test_trap_fork (0, G_TEST_TRAP_SILENCE_STDOUT | G_TEST_TRAP_SILENCE_STDERR))
-- 
1.7.7.3

