Subject: [PATCH] resources: compiler: Fix resources on big endian
 architectures
https://bugzilla.gnome.org/show_bug.cgi?id=673409
---

Index: glib2.0-2.32.1/gio/gresource.c
===================================================================
--- glib2.0-2.32.1.orig/gio/gresource.c	2012-04-13 15:39:41.000000000 +0200
+++ glib2.0-2.32.1/gio/gresource.c	2012-04-16 06:36:35.796998600 +0200
@@ -287,7 +287,7 @@
       free_path[path_len-1] = 0;
     }
 
-  value = gvdb_table_get_value (resource->table, path);
+  value = gvdb_table_get_raw_value (resource->table, path);
 
   if (value == NULL)
     {
@@ -305,6 +305,9 @@
                      &_flags,
                      &array);
 
+      _size = GUINT_FROM_LE (_size);
+      _flags = GUINT_FROM_LE (_flags);
+
       if (size)
         *size = _size;
       if (flags)
