From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 29 Jul 2019 17:26:20 +0200
Subject: gnetworkmonitornm: Arguments to g-signal's callback are const

Origin: upstream, 2.61.3, commit:109f247039696b92c3d696bee1d0377b7b4cb33a
---
 gio/gnetworkmonitornm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gio/gnetworkmonitornm.c b/gio/gnetworkmonitornm.c
index 52073fa..479653a 100644
--- a/gio/gnetworkmonitornm.c
+++ b/gio/gnetworkmonitornm.c
@@ -268,8 +268,8 @@ update_cached_property (GDBusProxy   *proxy,
 
 static void
 proxy_signal_cb (GDBusProxy        *proxy,
-                 gchar             *sender_name,
-                 gchar             *signal_name,
+                 const gchar       *sender_name,
+                 const gchar       *signal_name,
                  GVariant          *parameters,
                  GNetworkMonitorNM *nm)
 {
