From: Iain Lane <laney@debian.org>
Date: Tue, 18 Mar 2014 15:43:35 +0000
Subject: Skip test which performs some unreliable floating point comparisons

Bug: https://gitlab.gnome.org/GNOME/glib/issues/820
Forwarded: no
---
 glib/tests/timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/tests/timer.c b/glib/tests/timer.c
index 4cd5fba..d5f5bbb 100644
--- a/glib/tests/timer.c
+++ b/glib/tests/timer.c
@@ -278,7 +278,7 @@ main (int argc, char *argv[])
 {
   g_test_init (&argc, &argv, NULL);
 
-  g_test_add_func ("/timer/basic", test_timer_basic);
+/*  g_test_add_func ("/timer/basic", test_timer_basic);*/
   g_test_add_func ("/timer/stop", test_timer_stop);
   g_test_add_func ("/timer/continue", test_timer_continue);
   g_test_add_func ("/timer/reset", test_timer_reset);
