From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 29 Jul 2019 17:27:19 +0200
Subject: tests: Remove unneeded ";" in G_DEFINE_INTERFACE_WITH_CODE

Origin: upstream, 2.61.3, commit:b80fcc6cb803ac724021e0b2c97f17bbe107591a
---
 gobject/tests/type.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gobject/tests/type.c b/gobject/tests/type.c
index 1abe482..c5db7e9 100644
--- a/gobject/tests/type.c
+++ b/gobject/tests/type.c
@@ -33,7 +33,7 @@ typedef struct {
 GType foo_get_type (void);
 
 G_DEFINE_INTERFACE_WITH_CODE (Foo, foo, G_TYPE_OBJECT,
-                              g_type_interface_add_prerequisite (g_define_type_id, bar_get_type ());)
+                              g_type_interface_add_prerequisite (g_define_type_id, bar_get_type ()))
 
 static void
 foo_default_init (FooInterface *iface)
