From: Xavier Claessens <xavier.claessens@collabora.com>
Date: Thu, 20 Sep 2018 09:36:20 -0400
Subject: Tests: Fix -Wformat-nonliteral causing build failure with clang

Origin: upstream, 2.59.0, commit:abac67c050c1e795035a1d691a87817b7f41dab7
---
 tests/unicode-encoding.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/unicode-encoding.c b/tests/unicode-encoding.c
index 8ac4f95..80427bd 100644
--- a/tests/unicode-encoding.c
+++ b/tests/unicode-encoding.c
@@ -9,6 +9,8 @@
 
 static gint exit_status = 0;
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 static void
 croak (char *format, ...)
 {
@@ -32,6 +34,7 @@ fail (char *format, ...)
 
   exit_status |= 1;
 }
+#pragma GCC diagnostic pop
 
 typedef enum
 {
