From: =?utf-8?b?0KDRg9GB0LvQsNC9INCY0LbQsdGD0LvQsNGC0L7Qsg==?=
 <lrn1986@gmail.com>
Date: Thu, 20 Sep 2018 07:26:49 +0000
Subject: tests W32: ifdef a variable not used on Windows

Origin: upstream, 2.59.0, commit:3ae1244ba10de8a0070613b0377a2deabeee1822
---
 tests/mapping-test.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/mapping-test.c b/tests/mapping-test.c
index 27c6d83..ad776fa 100644
--- a/tests/mapping-test.c
+++ b/tests/mapping-test.c
@@ -191,7 +191,9 @@ test_child_private (gchar *argv0)
   gsize len;
   gchar *child_argv[4];
   GPid  child_pid;
+#ifndef G_OS_WIN32
   GMainLoop *loop;
+#endif
   gchar pid[100];
   
 #ifdef G_OS_WIN32
